### PR TITLE
Next gen

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Next gen for everyone.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -16,25 +16,25 @@ var app = express();
...
+app.use(bodyParser.urlencoded({extended: false}));
 app.use(cookieParser());
 app.use(session({
   secret: 'secret',
   name: 'browser-calls',
-  cookie: { maxAge: 60000 },
+  cookie: {maxAge: 60000},
   resave: true,
-  saveUninitialized: true
+  saveUninitialized: true,
```
Comment (acamino) [inline app.js]: Out of curiosity, why do you put a trailing comma in every literal object?

Comment (juancarlospaco) [inline app.js]: JavaScript Linter best practices says so... :confused:

File: app.js [hunk 2]
```diff
@@ -69,8 +69,9 @@ app.use(function(err, req, res, next) {
   res.status(err.status || 500);
   res.render('error', {
     message: err.message,
-    error: {}
+    error: {},
   });
 });
 
+
```
Comment (acamino) [inline app.js]: We should use a single blank line between functions definitions. Adding multiple blank lines is permitted but not required or encouraged. Would you mind removing the extra blank lines? Unless you have a different point of view on this matter.

File: config.js [hunk 3]
```diff
@@ -29,3 +29,8 @@ cfg.mongoUrlTest = process.env.MONGO_URL_TEST || defaultMongoUrlTest;
 
 // Export configuration object
 module.exports = cfg;
+
+
+if (!cfg.twilioAccountSid || !cfg.twilioAuthToken) {
```
Comment (acamino) [inline config.js]: This looks good. I'm wondering if you would consider to include `dotenv-safe` package that @joliveros mentioned?

File: models/order.js [hunk 4]
```diff
@@ -2,39 +2,34 @@ var mongoose = require('mongoose');
...
-  notificationStatus : { type: String, default: 'None' },
+  status: {type: String, default: 'Ready'},
+  notificationStatus: {type: String, default: 'None'},
 });
 
-OrderSchema.methods.sendSmsNotification = function (message, statusCallback , callback) {
+OrderSchema.methods.sendSmsNotification = function(message, statusCallback, callback) {
 
-  var client = new twilio.RestClient(cfg.twilioAccountSid, cfg.twilioAuthToken);
+  var client = twilio(config.accountSid, config.authToken);
```
Comment (acamino) [inline models/order.js]: This won't work because the `config` variable is not defined. Which means that also some tests will fail. Surprisingly Travis is telling that the build is green, but this reveals a subtle problem, the build is not running the test suite.

Could you please use `cfg` instead of `config` or the other way around. 
One quick fix for having the tests working in the right way is to include a test script in the `package.json` file.

I'd love to hear your thoughts on this.

File: app.js (hunk 1 above)
Comment (joliveros) [inline app.js]: I believe this is to avoid having to replace the line in git if you ever have to add another item to the object just because a comma was added to the end of the line.

File: config.js (hunk 3 above)
Comment (juancarlospaco) [inline config.js]: I agree.
Code where write near the same time I read it @joliveros comment about `dotenv-safe` ;P

File: .travis.yml [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
   global:
-    - export TWILIO_ACCOUNT_SID=AC2XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-    - export TWILIO_AUTH_TOKEN=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-    - export TWILIO_NUMBER=+15551230987
-    - export MONGO_URL_TEST=mongodb://localhost/appointment-reminders-node-test
+    - TWILIO_ACCOUNT_SID=AC2XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    - TWILIO_AUTH_TOKEN=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    - TWILIO_NUMBER=+15551230987
+    - MONGO_URL_TEST=mongodb://localhost/appointment-reminders-node-test
+    - MONGO_URL=mongodb://localhost/appointment-reminders-node-test
```
Comment (acamino) [inline .travis.yml]: There is no need to add this `MONGO_URL` env variable. Could you please remove it?

File: app.js (hunk 1 above)
Comment (acamino) [inline app.js]: That's an interesting reason @joliveros, is about having a clean diff. If you have to add or remove a line in an object literal, the _diff_ will only contain the line you add or remove, e.g.:

When removing a line:

```diff
{
   foo: 'foo',
-  bar: 'bar',
 }
```

When adding a line:

```diff
 {
   foo: 'foo',
   bar: 'bar',
+  tar: 'tar',
 }
```

The same example when you don't have a trailing comma in the last element of your literal object and you want to add a new key.

```diff
 {
   foo: 'foo',
-  bar: 'bar'
+  bar: 'bar',
+  tar: 'tar'
 }
```

😉

File: package.json [hunk 6]
```diff
@@ -3,7 +3,8 @@
   "version": "1.0.0",
   "private": true,
   "scripts": {
-    "start": "node ./bin/www"
+    "start": "node ./bin/www",
+    "test": "NODE_ENV=test mocha test"
```
Comment (acamino) [inline package.json]: @juancarlospaco because you introduced the _dotenv-safe_ dependency there is no step in where the user exports the env variables.

You have to explicitly set the env variables for testing purposes here. Like this:

```
"test": "NODE_ENV=test TWILIO_ACCOUNT_SID=AC2XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX TWILIO_AUTH_TOKEN=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX TWILIO_NUMBER=+15551230987 MONGO_URL_TEST=mongodb://localhost/appointment-reminders-node-test ./node_modules/mocha/bin/mocha test"
```

File: .env.example [hunk 7]
```diff
@@ -9,5 +9,4 @@ TWILIO_AUTH_TOKEN=Your-Twilio-Auth-Token
 TWILIO_NUMBER=Your-Twilio-Phone-Number
 
 # Your mongo db urls
-MONGO_URL=Mongo-Url
```
Comment (acamino) [inline .env.example]: Why did you remove this env variable? Please take a look at https://github.com/TwilioDevEd/eta-notifications-node/blob/15d4900572e3fdc53043684d58aa8e0362eedd58/config.js#L27

Comment (juancarlospaco) [inline .env.example]: it caused the build to fail: https://travis-ci.org/TwilioDevEd/eta-notifications-node/jobs/177705048

**Comment by acamino** [comment]
Thanks! This one is ready to get merged.

**Comment by juancarlospaco** [comment]
Looks good for me.

File: config.js [hunk 8]
```diff
@@ -17,10 +19,8 @@ module.exports = {
 
   // A Twilio number you control - choose one from:
   // Specify in E.164 format, e.g. "+16519998877"
-  twilioPhoneNumber: process.env.TWILIO_PHONE_NUMBER,
+  twilioPhoneNumber: process.env.TWILIO_NUMBER,
 
   // MongoDB connection string - MONGO_URL is for local dev,
   mongoUri: process.env.MONGO_URI,
-  mongoUriTest: process.env.MONGO_URI_TEST,
```
Comment (acamino) [inline config.js]: +1

File: config.js [hunk 9]
```diff
@@ -1,3 +1,5 @@
+require('dotenv-safe').load();
+
 // Fallback values
 var defaultMongoUrl = 'mongodb://localhost:27017/eta-notifications-node';
```
Comment (acamino) [inline config.js]: Why to we keep the fallback values? - Looks like dead code. 💀

File: models/order.js [hunk 10]
```diff
@@ -21,13 +24,10 @@ OrderSchema.methods.sendSmsNotification = function(message, statusCallback, call
...
-    .then((message) => console.log('Message sent to ' + message.customerPhoneNumber))
-    .catch((error) => console.log(error));
-
-  if (callback) {
-    callback.call(self);
-  }
+  return client.messages.create(options)
+    .then((message) => {
+      console.log('Message sent to ' + message.to);
+    })
```
Comment (acamino) [inline models/order.js]: Did you consider adding a `catch` method to deal with rejected cases? Moreover, seems like you missed a semicolon. 😉

File: config.js (hunk 9 above)
Comment (hortega) [inline config.js]: Good catch!

File: models/order.js (hunk 10 above)
Comment (hortega) [inline models/order.js]: We added one but in routes/orders.js, there we can send a 500 response to the requester
https://github.com/TwilioDevEd/eta-notifications-node/blob/next-gen/routes/orders.js#L35

**Comment by hortega** [comment]
Thanks @acamino !

File: models/order.js (hunk 10 above)
Comment (acamino) [inline models/order.js]: Sweet!

**Comment by acamino** [comment]
LGTM! This one is ready!
